src/pages/_app.tsx

```diff
@@ -33,6 +34,23 @@ const { ConsentManager, openConsentManager } = createConsentManager({
 export default function App({ Component, pageProps, layoutProps }) {
   useAnchorLinkAnalytics()
 
+  /**
+   * TODO: this will be temporary and last for as long as we have to manually
+   * hydrate product data from page components. Ideally, we won't have to do
+   * this we use more dynamic routes.
+   */
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+  const isDevDotPage =
+    !currentPath.startsWith(`/_proxied-dot-io`) &&
+    !currentPath.startsWith(`/swingset`)
+  const isNotHomePage = currentPath !== '/'
+  const isMissingProductData = !pageProps.product
+  if (isDevDotPage && isNotHomePage && isMissingProductData) {
+    throw new Error(
+      `\`product\` property is missing from \`pageProps\` prop on path: \`${currentPath}\`. Product data is needed to hydrate \`CurrentProductContext\`.`
+    )
+  }
```

Comment:
what do you think about gating this behind `if (process.env.HASHI_ENV !== 'production')`? Realistically we shouldn't need to do this check in production once the app has been built and the pages have been rendered.

Reply:
@BRKalow do you know what `HASHI_ENV` is during the Vercel deploy check? I think it's `'preview'`, but I'm not 100% sure. If it is `'preview'`, I think this'll be a good add!

Reply:
Testing it out with: https://github.com/hashicorp/dev-portal/pull/222/commits/b1ea71ac69ec0f33c4092cb4dc2d751ac72348c6. Will fix the line afterwards. 😊

Reply:
Yep, it is `preview` for deploy previews, production builds on main have `production` 👍